src/hooks/useBalance/useCosmosBalance.tsx

```diff
@@ -1,17 +1,46 @@
-import { useCallback, useEffect, useMemo, useState } from 'react'
+import { useCallback, useEffect, useState } from 'react'
 import { StargateClient } from '@cosmjs/stargate'
+import { useQuery } from '@tanstack/react-query'
+import { useTabActive } from '../useTabActive/useTabActive'
 import type { TUseBalanceOptions } from './types'
 import { isCosmosWallet } from '@/utils/wallet'
 import { getNetworkById, rpcMapping } from '@/networks'
 
-const SECONDS_BEFORE_NEXT_UPDATE = 10
+const balanceFetcher = (options: TUseBalanceOptions, network: ReturnType<typeof getNetworkById>, client: StargateClient) => {
+  const { address } = options
+
+  if (!isCosmosWallet(options) || !address || !client || !network) {
```

Comment:
I would suggest to use `isCosmosWallet/isEvmWallet/isSolWallet` check on some upper layer. Maybe somewhere in  `PickBalanceProvider`